File: backend/spring-boot/src/main/java/org/bootstrapbugz/api/auth/jwt/redis/model/UserBlacklist.java
package org.bootstrapbugz.api.auth.jwt.redis.model;

import java.io.Serial;
import java.io.Serializable;
import java.time.Instant;
import java.time.temporal.ChronoUnit;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.redis.core.RedisHash;
import org.springframework.data.redis.core.TimeToLive;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@RedisHash(value = "UserBlacklist")
public class UserBlacklist implements Serializable {
  @Serial private static final long serialVersionUID = 8334740937644612692L;

  @Id private Long userId;

  @Builder.Default private Instant updatedAt = Instant.now().truncatedTo(ChronoUnit.SECONDS);

  @TimeToLive private long timeToLive;
}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/auth/jwt/redis/model/UserBlacklist.kt
package org.bootstrapbugz.api.auth.jwt.redis.model

import java.io.Serial
import java.io.Serializable
import java.time.Instant
import java.time.temporal.ChronoUnit
import org.springframework.data.annotation.Id
import org.springframework.data.redis.core.RedisHash
import org.springframework.data.redis.core.TimeToLive

@RedisHash(value = "UserBlacklist")
class UserBlacklist(
  @Id val userId: Long? = null,
  @TimeToLive val timeToLive: Long = 0,
  val updatedAt: Instant = Instant.now().truncatedTo(ChronoUnit.SECONDS)
) : Serializable {

  companion object {
    @Serial private val serialVersionUID = 8334740937644612692L
  }
}
